js/src/Pie.js

```diff
@@ -303,7 +265,8 @@ var Pie = mark.Mark.extend({
             })
             .each(function(d) {
                 this._current = d;
-            });
+            }).on("click", function(d, i) {
```

Comment:
I've moved this here as a bugfix, and also for correctness. From my understanding we would only want to add the event handler to new elements.